src/kemal/middleware/filters.cr

```diff
@@ -74,7 +74,10 @@ module Kemal::Middleware
     property block : HTTP::Server::Context -> String
 
     def initialize(&block : HTTP::Server::Context -> _)
-      @block = ->(context : HTTP::Server::Context) { block.call(context).to_s }
+      @block = ->(context : HTTP::Server::Context) do 
+        block.call(context).to_s.to_json if context.request.headers["Content-Type"]? == "application/json"
```

Comment:
Hm, strange. But the error points to `handler.call(context).to_json`, while in your code you have `block.call(context).to_s.to_json`... maybe you did the change in another place? In any case maybe implicitly doing things is not a good idea after all, I don't know. Maybe changing the docs is simpler for now :-)